### PR TITLE
Add -NoClobber functionality to Install-PSResource

### LINKED ISSUE #203: Install-PSResource -NoClobber error message doesn't show name conflicts
When `-NoClobber` is specified and a command conflict is found the command list is not expanded to show the individual names but rather the type name is shown.

```powershell
C:\Users\Thomas> Install-PSResource acme-ps -NoClobber

Untrusted repository
You are installing the modules from an untrusted repository. If you trust this repository, change its Trusted value by
running the Set-PSResourceRepository cmdlet. Are you sure you want to install the PSresource from 'PSGallery' ?
[Y] Yes  [A] Yes to All  [N] No  [L] No to All  [S] Suspend  [?] Help (default is "N"): y
Install-PSResource: Command(s) with name(s) 'System.Linq.Enumerable+WhereEnumerableIterator`1[System.Management.Automation.PSObject]' is already available on this system. Installing 'ACME-PS' may override the existing command. If you still want to install 'ACME-PS', remove the -NoClobber parameter.
```

### LINKED ISSUE #204: Install-PSResource with NoClobber produces command conflict when there is none
During `Install-PSResource` with `-NoClobber` produces an error for a command conflict when there is none.

Steps to reproduce.

```powershell
Get-Command *acme* # returns nothing

Install-PSResource acme-ps -TrustRepository -NoClobber -Verbose
VERBOSE: Current user scope installation path: C:\Users\Thomas\Documents\WindowsPowerShell
VERBOSE: All users scope installation path: C:\Program Files\WindowsPowerShell
VERBOSE: Scope is: CurrentUser
Install-PSResource: Command(s) with name(s) 'System.Linq.Enumerable+WhereEnumerableIterator`1[System.Management.Automation.PSObject]' is already available on this system. Installing 'ACME-PS' may override the existing command. If you still want to install 'ACME-PS', remove the -NoClobber parameter.

### PR DESCRIPTION
**Pull request by alerickson**
# PR Summary

Add functionality to Install-PSResource for -NoClobber parameter

## PR Context

Resolves #203 
Resolves #204 

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary.

### CODE REVIEW

File: src/code/InstallHelper.cs [hunk 1]
```diff
@@ -538,6 +544,55 @@ private bool CallAcceptLicense(PSResourceInfo p, string moduleManifest, string t
             return success;
         }
 
+        private bool DetectClobber(string pkgName, string tempDirNameVersion, Hashtable parsedMetadataHashtable)
+        {
+            // Get installed modules, then get all possible paths
+            bool foundClobber = false;
+            GetHelper getHelper = new GetHelper(_cmdletPassedIn);
+            IEnumerable<PSResourceInfo> pkgsAlreadyInstalled = getHelper.FilterPkgPaths(new string[] { "*" }, VersionRange.All, _pathsToSearch);
```
Comment (PaulHigin) [inline src/code/InstallHelper.cs]: 'FilterPkgPaths' method name is misleading.  It is not filtering package paths, but instead is getting PSResourceInfo information on every installed resource indicated by the directory path.
Should be 'GetPkgInfoFromPaths()' or something similar.

File: src/code/InstallHelper.cs [hunk 2]
```diff
@@ -538,6 +544,55 @@ private bool CallAcceptLicense(PSResourceInfo p, string moduleManifest, string t
...
 
+        private bool DetectClobber(string pkgName, string tempDirNameVersion, Hashtable parsedMetadataHashtable)
+        {
+            // Get installed modules, then get all possible paths
+            bool foundClobber = false;
+            GetHelper getHelper = new GetHelper(_cmdletPassedIn);
+            IEnumerable<PSResourceInfo> pkgsAlreadyInstalled = getHelper.FilterPkgPaths(new string[] { "*" }, VersionRange.All, _pathsToSearch);
+            // user parsed metadata hash
+            List<string> listOfCmdlets = new List<string>();
+            foreach (var cmdlet in parsedMetadataHashtable["CmdletsToExport"] as object[])
```
Comment (PaulHigin) [inline src/code/InstallHelper.cs]:
```suggestion
            foreach (var cmdletName in parsedMetadataHashtable["CmdletsToExport"] as object[])
```

File: src/code/InstallHelper.cs [hunk 3]
```diff
@@ -538,6 +544,55 @@ private bool CallAcceptLicense(PSResourceInfo p, string moduleManifest, string t
...
+            {
+                listOfCmdlets.Add(cmdlet as string);
```
Comment (PaulHigin) [inline src/code/InstallHelper.cs]:
```suggestion
                listOfCmdlets.Add(cmdletName as string); 

```

File: src/code/InstallHelper.cs (hunk 2 above)
Comment (PaulHigin) [reply src/code/InstallHelper.cs]: We also need to include module 'FunctionsToExport' in list.

File: src/code/InstallHelper.cs [hunk 4]
```diff
@@ -538,6 +544,55 @@ private bool CallAcceptLicense(PSResourceInfo p, string moduleManifest, string t
...
+            }
+
+            foreach (var pkg in pkgsAlreadyInstalled)
+            {
+                List<string> duplicateCmdlets = new List<string>();
+                List<string> duplicateCmds = new List<string>();
+                // See if any of the cmdlets or commands in the pkg we're trying to install exist within a package that's already installed
+                if (pkg.Includes.Cmdlet != null && pkg.Includes.Cmdlet.Any())
+                {
+                    duplicateCmdlets = listOfCmdlets.Where(cmdlet => pkg.Includes.Cmdlet.Contains(cmdlet)).ToList();
```
Comment (PaulHigin) [inline src/code/InstallHelper.cs:595]: We need to do case insensitive string compare.

**Review by PaulHigin — CHANGES_REQUESTED** [review]

**Review by anamnavi — APPROVED** [review]
LGTM

**Pull request merged**